app/src/main/java/eu/kanade/tachiyomi/data/sync/SyncManager.kt

```diff
@@ -0,0 +1,209 @@
+ *
+ * @property context The application context.
+ */
+class SyncManager(
+    private val context: Context,
+    private val handler: DatabaseHandler = Injekt.get(),
+    private val syncPreferences: SyncPreferences = Injekt.get(),
+    private var json: Json = Json {
+        encodeDefaults = true
+        ignoreUnknownKeys = true
+    },
+    private val getFavorites: GetFavorites = Injekt.get(),
+    private val getCategories: GetCategories = Injekt.get(),
+
+) {
+    private val backupManager: BackupManager = BackupManager(context)
+    private val notifier: SyncNotifier = SyncNotifier(context)
+
+    enum class SyncService(val value: Int) {
+        NONE(0),
+        SYNCYOMI(1),
+        ;
+
+        companion object {
+            fun fromInt(value: Int) = values().firstOrNull { it.value == value } ?: NONE
+        }
+    }
+
+    /**
+     * Syncs data with a sync service.
+     *
+     * This function retrieves local data (favorites, manga, extensions, and categories)
+     * from the database using the BackupManager, then synchronizes the data with a sync service.
+     */
+    suspend fun syncData() {
+        val databaseManga = getAllMangaFromDB()
+
+        logcat(LogPriority.DEBUG) { "Mangas to sync: $databaseManga" }
+        val backup = Backup(
+            backupManager.backupMangas(databaseManga, BACKUP_ALL),
+            backupManager.backupCategories(BACKUP_ALL),
+            emptyList(),
+            backupManager.backupExtensionInfo(databaseManga),
+        )
+
+        // Create the SyncStatus object
+        val syncStatus = SyncStatus(
+            lastSynced = Instant.now().toString(),
+            status = "completed",
+        )
+
+        // Create the Device object
+        val device = SyncDevice(
+            id = syncPreferences.deviceID().get(),
+            name = syncPreferences.deviceName().get(),
+        )
+
+        // Create the SyncData object
+        val syncData = SyncData(
+            sync = syncStatus,
+            backup = backup,
+            device = device,
+        )
+
+        // Handle sync based on the selected service
+        val syncService = when (val syncService = SyncService.fromInt(syncPreferences.syncService().get())) {
+            SyncService.SYNCYOMI -> {
+                SyncYomiSyncService(
+                    context,
+                    json,
+                    syncPreferences,
+                    notifier,
+                )
+            }
+
+            else -> {
+                logcat(LogPriority.ERROR) { "Invalid sync service type: $syncService" }
+                null
+            }
+        }
+
+        val remoteBackup = syncService?.doSync(syncData)
+
+        if (remoteBackup != null) {
+            val (filteredFavorites, nonFavorites) = filterFavoritesAndNonFavorites(remoteBackup)
+            updateNonFavorites(nonFavorites)
+            SyncHolder.backup = backup.copy(backupManga = filteredFavorites)
```

Comment:
This is wrong by the way, It won't restore the categories and just place all the manga in the default categories instead.

I have updated it in the master branch and fixed it @AriaMoradi but basically we want to copy the rest of the data as well.

```suggestion
            SyncHolder.backup = backup.copy(
                backupManga = filteredFavorites,
                backupCategories = remoteBackup.backupCategories,
                backupSources = remoteBackup.backupSources,
                backupBrokenSources = remoteBackup.backupBrokenSources,
            )
```